src/compiler/checker.ts

```diff
@@ -15631,22 +15642,35 @@ namespace ts {
             //  declare function f(a: { xa: number; xb: number; });

```

Comment:
comments above here need to be updated and typos fixed (specifically, that &rarr; than in one case)